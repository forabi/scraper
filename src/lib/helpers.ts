import * as fs from 'fs';
import * as got from 'got';
import { promisify } from 'util';
import * as _glob from 'glob';

export const readFile = promisify(fs.readFile);
export const writeFile = promisify(fs.writeFile);
export const readDir = promisify(fs.readdir);

export const glob = async (pattern: string, options: _glob.IOptions = {}) => {
  return new Promise<string[]>((resolve, reject) => {
    _glob(pattern, options, (error, result) => {
      if (error) {
        reject(error);
      } else {
        resolve(result);
      }
    });
  });
};

export const readJsonFile = async <T = any>(file: string) =>
  readFile(file, 'utf8').then(JSON.parse) as Promise<T>;

export function replaceSmartQuotes(str: string) {
  // prettier-ignore
  return str.replace(/[‘’]/g, '\'').replace(/[“”]/g, '"');
}

export async function fetchPageAsHtml(url: string) {
  const response = await got(url, {
    headers: {
      Accept: 'text/html',
    },
  });

  return response.body;
}
