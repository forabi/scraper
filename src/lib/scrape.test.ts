import { scrapeHtml, isResultWithContent, Result, isPiece } from './scrape';
import * as path from 'path';
import * as bluebird from 'bluebird';

import { readDir, readFile } from './helpers';

const types = ['complete', 'incomplete', 'stub'];

describe('works for all post types', async () => {
  for (const type of types) {
    describe(`works for ${type} posts`, async () => {
      let files: Array<{ filename: string; result: Result }>;

      beforeAll(async () => {
        // tslint:disable-next-line:await-promise
        files = await bluebird.map(
          readDir(path.join('fixtures', 'html', type)),
          async filename => {
            const html = await readFile(
              path.join('fixtures', 'html', type, filename),
              'utf8',
            );

            return {
              filename,
              result: await scrapeHtml(html),
            };
          },
        );
      });

      it('matches snapshot', () => {
        for (const { result, filename } of files) {
          expect(result).toMatchSnapshot(filename);
        }
      });

      it('every post has a name', () => {
        for (const { result } of files) {
          expect(result.name).toBeDefined();
          expect(result.name.length).toBeGreaterThan(0);
        }
      });

      it('posts that are not stubs must have some content', () => {
        for (const { result } of files) {
          if (isResultWithContent(result)) {
            expect(result.content.length).toBeGreaterThan(0);
            const piece = result.content[0];
            if (isPiece(piece)) {
              expect(piece.text.length).toBeGreaterThan(0);
            }
          }
        }
      });
    });
  }
});
