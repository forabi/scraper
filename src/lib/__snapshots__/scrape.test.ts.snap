// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`al-pacino 1`] = `
Object {
  "content": Array [
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "Al Pacino was born in Manhattan, New York and grew up in The Bronx, New York.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "Pacino is the son of full-on Italians and was raised as such, which includes Catholicism. But it seems to have been only in name that Pacino was Catholic. He said:",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Al Pacino’s Remembrance of Things (and Flings) Past",
      "sourceUrl": "http://www.moviecrazed.com/outpast/alpacdowney.html",
      "text": "My parents were Italian, and they had me baptized, but I was not raised in the Catholic tradition. I come from a broken home.",
      "type": "quote",
    },
    Object {
      "sourceTitle": "Al Pacino’s Remembrance of Things (and Flings) Past",
      "sourceUrl": "http://www.moviecrazed.com/outpast/alpacdowney.html",
      "text": "Pacino is probably referring to the fact that his parents divorced when he was two-years-old and, contrary to Catholic tradition, his father was married four times.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Al Pacino’s Biography",
      "sourceUrl": "http://www.oceleb.com/al-pacino/biography.html",
      "text": "So, Pacino indicates that he's not really a Catholic. Then what is he? This is where things get difficult. Much like his fellow iconic, mobster movie stars Marlon Brando and Robert De Niro, Pacino won't talk about it in interviews. Not talking about his religion might be a precondition to giving an interview. Some sources cite him as an agnostic,",
      "type": "sentence",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "which makes sense. It's the easy way out of religious commitment. He did once give his philosophy of mankind–which is cryptic and almost meaningless:",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Quotes",
      "sourceUrl": "http://velvet_peach.tripod.com/apacinoquotes.html",
      "text": "My philosophy? Man is a little bit better than his reputation, and a little bit worse.",
      "type": "quote",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "He's just the kind of guy who likes to remain a mystery.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "Pacino politics",
      "type": "heading",
    },
    Object {
      "sourceTitle": "Al Pacino’s Federal Campaign Contribution Report",
      "sourceUrl": "http://www.newsmeat.com/celebrity_political_donations/Al_Pacino.php",
      "text": "Al is only slightly less secretive about his political views. But again, it's probably something he doesn't allow interviewers to question him about. He supports Democrats (at least one) financially. Pacino has only given money to one political candidate–Democrat Ralph Waite–but he's contributed to Waite's campaign four different times, to the tune of $4,000.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "President Obama to Award National Medal of Arts to Al Pacino",
      "sourceUrl": "http://www.hollywoodreporter.com/news/al-pacino-president-obama-national-medal-of-arts-289385",
      "text": "Democrats support Pacino as well. He was given the National Medal of Arts by President Barack Obama in 2011.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "Pacino indicates his liberal view of sexual orientation, indicating confusion regarding any sort of intolerance. He said:",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Al Pacino – Biography.",
      "sourceUrl": "http://www.imdb.com/name/nm0000199/bio",
      "text": "I don't understand the hatred and fear of gays and bisexuals and lesbians…it's a concept I honestly cannot grasp. To me, it's not who you love…a man, a woman, what have you…it's the fact THAT you love. That is all that truly matters.",
      "type": "quote",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "Maybe that's the part where mankind is a little worse than his reputation.",
      "type": "sentence",
    },
  ],
  "notablePerson": Object {
    "author": "Tom Kershaw",
    "lastUpdatedOn": 2012-08-29T21:00:00.000Z,
    "name": "Al Pacino",
    "relatedPeople": Array [
      Object {
        "name": "George Clooney",
        "slug": "george-clooney",
      },
      Object {
        "name": "Sean Penn",
        "slug": "sean-penn",
      },
      Object {
        "name": "Robert Redford",
        "slug": "robert-redford",
      },
      Object {
        "name": "Emily Deschanel",
        "slug": "emily-deschanel",
      },
    ],
    "summary": Object {
      "politicalViews": "Pacino has supported Democrats financially and expressed a liberal view toward sexual orientation.",
      "religion": "Pacino was raised an Italian Catholic, but it wasn't a powerful force in his family. Now, he's probably an agnostic.",
    },
    "tags": Array [
      "actor",
      "agnostic",
      "american",
      "catholic",
      "democrat",
      "director",
      "italian",
      "liberal",
      "producer",
      "white",
    ],
  },
}
`;

exports[`george-clooney 1`] = `
Object {
  "content": Array [
    Object {
      "sourceTitle": "The Politics of George Clooney, Actor and Liberal Political Activist",
      "sourceUrl": "http://usliberals.about.com/od/celebrityactivists/a/GClooney.htm",
      "text": "George Clooney was raised in Georgetown, Kentucky in a devout Catholic family.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "Clooney speaks highly of his Catholic roots, saying:",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Interview: George Clooney",
      "sourceUrl": "http://www.exacteditions.com/exact/browse/397/440/2440/3/16",
      "text": "I used to go to confession every week….I grew up with a great sense of structure and respect.",
      "type": "quote",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "Clearly, Clooney has outgrown his childhood religion and now, has no problem telling the world he's an Agnostic. In fact, some quotes indicate that Clooney might be more of an atheist:",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "George Clooney.",
      "sourceUrl": "http://www.celebatheists.com/wiki/George_Clooney",
      "text": "I don't believe in Heaven and Hell. I don't know if I believe in God. All I know is that as an individual, I won't allow this life — the only thing I know to exist — to be wasted.",
      "type": "quote",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "Still, he's very careful to stress that he has no problem with those with religious beliefs, saying:",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Transcripts",
      "sourceUrl": "http://transcripts.cnn.com/TRANSCRIPTS/0602/16/lkl.01.html",
      "text": "Whatever anybody believes as long as it doesn't hurt anybody else, it's fair enough, and works, and I think, is real, and matters. I don't happen to have those beliefs, as much, you know, I don't believe in those things.",
      "type": "quote",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "Celebrity superhero to the rescue",
      "type": "heading",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "Politically, Clooney has been very active throughout his career. He calls himself a liberal and wonders why society has turned on liberals, saying:",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "The Politics of George Clooney, Actor and Liberal Political Activist",
      "sourceUrl": "http://usliberals.about.com/od/celebrityactivists/a/GClooney.htm",
      "text": "I'm a liberal. I'm confused when that became a bad word…the liberal movement morally, you know, has stood on the right side of an awful lot of issues. We thought that blacks should be allowed to sit at the front of the bus and women should be able to vote, McCarthy was wrong, Vietnam was a mistake.",
      "type": "quote",
    },
    Object {
      "sourceTitle": "George Clooney – Address to the United Nations Security Council on Darfur",
      "sourceUrl": "http://www.americanrhetoric.com/speeches/georgeclooneyunitednations.htm",
      "text": "Clooney was so disturbed by genocide efforts in Darfur that he addressed the United Nations Security Council about it, hoping for more intervention.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "It's not surprising Clooney turned his fame as an actor towards political causes. His father was a journalist who went after corrupt politicians in his time. Clooney said:",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "The Politics of George Clooney, Actor and Liberal Political Activist",
      "sourceUrl": "http://usliberals.about.com/od/celebrityactivists/a/GClooney.htm",
      "text": "My dad, who went after OPEC for raising gas prices and Gerald Ford for pardoning Richard Nixon, believed it's your responsibility—not just your right—to question authority.",
      "type": "quote",
    },
  ],
  "notablePerson": Object {
    "author": "Tom Kershaw",
    "lastUpdatedOn": 2012-02-26T22:00:00.000Z,
    "name": "George Clooney",
    "relatedPeople": Array [
      Object {
        "name": "Seth MacFarlane",
        "slug": "seth-macfarlane",
      },
      Object {
        "name": "Al Pacino",
        "slug": "al-pacino",
      },
      Object {
        "name": "Robert Redford",
        "slug": "robert-redford",
      },
      Object {
        "name": "Robert De Niro",
        "slug": "robert-de-niro",
      },
    ],
    "summary": Object {
      "politicalViews": "Clooney is a proud liberal and has lent his star power to many causes. He's spoken out against war, poverty, environmental issues, and more.",
      "religion": "Clooney was raised a Catholic, but has since proclaimed himself an agnostic, bordering on atheism.",
    },
    "tags": Array [
      "actor",
      "agnostic",
      "atheist",
      "catholic",
      "democrat",
      "director",
      "liberal",
      "producer",
      "white",
    ],
  },
}
`;

exports[`robert-redford 1`] = `
Object {
  "content": Array [
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "Robert Redford was born in Santa Monica, California and raised in Van Nuys, California.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Robert Redford – Biography",
      "sourceUrl": "http://www.imdb.com/name/nm0000602/bio",
      "text": "It is unclear if religion was present much during his childhood, but considering his Irish/Scottish/English heritage,",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Robert Redford: My Childhood of Crime",
      "sourceUrl": "http://www.express.co.uk/news/showbiz/259284/Robert-Redford-My-childhood-of-crime",
      "text": "it's reasonable to assume some sort of Christian background. However, as a youth, Redford was involved in street gangs, petty theft and general hooliganism,",
      "type": "sentence",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "so it seems unlikely he was devoted to any formal religious/moral framework.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "As an adult, Redford is wary of religion. He said:",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Robert Redford",
      "sourceUrl": "http://www.celebatheists.com/wiki/Robert_Redford",
      "text": "I don't believe in organized religion, because I don't believe people should be organized in how they think, in what they believe.",
      "type": "quote",
    },
    Object {
      "sourceTitle": "Robert Redford",
      "sourceUrl": "http://www.celebatheists.com/wiki/Robert_Redford",
      "text": "But he does seem to, at some point in his life, have taken up a spiritual quest. Redford has said he has \\"explored every religion, some very deeply\\" but that there's \\"there's not one philosophy that can satisfy [him].\\"",
      "type": "sentence",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "And when it comes to the great cosmological questions, like \\"What happens after we die,\\" Redford is skeptical:",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Robert Redford",
      "sourceUrl": "http://www.celebatheists.com/wiki/Robert_Redford",
      "text": "Is there an afterlife? As far as I know, this is it. It's all we've got. You take your opportunities and you go for it.",
      "type": "quote",
    },
    Object {
      "sourceTitle": "Religion of Top Box Office Stars (Actors)",
      "sourceUrl": "http://www.adherents.com/movies/actor_boxoffice_stars.html",
      "text": "I'd say this puts Redford in the agnostic category, perhaps bordering on atheist. He did, however, marry a Mormon woman",
      "type": "sentence",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "and lives and works in Utah (the seat of Mormonism), which has played into his political views–which we'll get into later.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "How about now?",
      "type": "heading",
    },
    Object {
      "sourceTitle": "Actor and NRDC Board Member Robert Redford: Join Hands Around the White House on November 6 to Tell President Obama No to Keystone XL Tar Sands Pipeline",
      "sourceUrl": "http://switchboard.nrdc.org/blogs/eshope/actor_and_nrdc_board_member_ro.html",
      "text": "Redford is a highly political person–an activist even–whose primary focus is environmentalism. He sits on the board of directors of the National Resources Defense Council",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Robert Redford",
      "sourceUrl": "http://www.foreverwildfilm.com/robertredford.html",
      "text": "and has won numerous awards for his environmental activity.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Robert Redford",
      "sourceUrl": "http://www.foreverwildfilm.com/robertredford.html",
      "text": "As early as the 70s, Redford was pushing for legislation like the Clean Air Act and the Energy Conservation and Production Act,",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Actor and NRDC Board Member Robert Redford: Join Hands Around the White House on November 6 to Tell President Obama No to Keystone XL Tar Sands Pipeline",
      "sourceUrl": "http://switchboard.nrdc.org/blogs/eshope/actor_and_nrdc_board_member_ro.html",
      "text": "and he continues to protest projects that threaten the environment such as the Keystone XL Pipeline",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Robert Redford",
      "sourceUrl": "http://www.foreverwildfilm.com/robertredford.html",
      "text": "and coal plants in southern Utah.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Robert Redford’s Federal Campaign Contribution Report",
      "sourceUrl": "http://www.newsmeat.com/celebrity_political_donations/Robert_Redford.php",
      "text": "This does not necessarily translate to alignment with the Democrats. While a little more than half of Redford's nearly $50,000 in campaign contributions has gone to Democrats, he has supported Republicans as well.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "And in 2011, Redford penned a piece for the  in which he slammed the Obama administration for not sticking to its word regarding environmental issues. He wrote:",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Robert Redford: Is The Obama Administration Putting Corporate Profits Above Public Health",
      "sourceUrl": "http://www.huffingtonpost.com/robert-redford/keystone-epa-obama-2012_b_947515.html",
      "text": "One reason I supported President Obama is because he said we must protect clean air, water and lands. But what good is it to say the right thing unless you act on it?",
      "type": "quote",
    },
    Object {
      "sourceTitle": "Robert Redford",
      "sourceUrl": "http://www.celebatheists.com/wiki/Robert_Redford",
      "text": "But, Redford dislikes what he sees as the Republican Party's connection to religion and has complained about George W. Bush claiming to receive guidance from God while in office",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Robert Redford knows all about Mormons and Mitt Romney",
      "sourceUrl": "http://bompolitic.wordpress.com/2007/11/09/robert-redford-knows-all-about-mormons-and-mitt-romney/",
      "text": "as well as Mitt Romney using slick rhetoric he learned as a Mormon missionary to sway votes in his favor.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Redford Goes Ron Paul",
      "sourceUrl": "http://www.theamericanconservative.com/articles/redford-goes-ron-paul/",
      "text": "Also, Redford is also known to be an admirer of famous anarchist Edward Abbey.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Robert Redford’s ‘Lions for Lambs’ takes on Iraq War",
      "sourceUrl": "http://www.sfgate.com/entertainment/article/Robert-Redford-s-Lions-for-Lambs-takes-on-Iraq-3301489.php",
      "text": "So, ultimately, he's difficult to pin down. Redford seems to be the type of political mind that realizes solutions can't be found in one ideology vs. another–though his critiques of America's foreign policy,",
      "type": "sentence",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "Republicans and his commitment to environmentalism generally puts him in the liberal category.",
      "type": "sentence",
    },
  ],
  "notablePerson": Object {
    "author": "Tom Kershaw",
    "lastUpdatedOn": 2013-05-20T21:00:00.000Z,
    "name": "Robert Redford",
    "relatedPeople": Array [
      Object {
        "name": "George Clooney",
        "slug": "george-clooney",
      },
      Object {
        "name": "Al Pacino",
        "slug": "al-pacino",
      },
      Object {
        "name": "Sean Penn",
        "slug": "sean-penn",
      },
      Object {
        "name": "Paul Newman",
        "slug": "paul-newman",
      },
    ],
    "summary": Object {
      "politicalViews": "Redford is mostly liberal with his focus on environmentalism and critiques of American foreign policy and Republicans, but has contributed to politicians across the spectrum and has expressed admiration for the anarchist ideology.",
      "religion": "Redford is an agnostic, maybe an atheist.",
    },
    "tags": Array [
      "actor",
      "agnostic",
      "american",
      "atheist",
      "democrat",
      "director",
      "liberal",
      "producer",
      "white",
    ],
  },
}
`;

exports[`seth-macfarlane 1`] = `
Object {
  "content": Array [
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "Seth MacFarlane was born in Newburyport, Massachusetts. MacFarlane grew up Catholic and went to church, however, he was satirizing religion as young as 11 years old, when he was allowed to draw a comic strip for the local newspaper. He recounts this story:",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Seth MacFarlane: TV’s ‘Family Guy’ Makes Music, Too",
      "sourceUrl": "http://www.npr.org/templates/transcript/transcript.php?storyId=140946967",
      "text": "I did a strip that had a guy kneeling at the altar taking Communion and saying can I have fries with that? And to my 11-year-old brain that was comedy.",
      "type": "quote",
    },
    Object {
      "sourceTitle": "Seth MacFarlane on atheism and gay rights",
      "sourceUrl": "http://www.youtube.com/watch?v=w3pExZs9IJA",
      "text": "He is an atheist and rather outspoken about it.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "But when it comes to atheism, MacFarlane is more Ricky Gervais than Christopher Hitchens. Occasionally, he'll get serious but, for the most part, MacFarlane is funny about it and doesn't seem to care to get into lengthy, theological or philosophical discussions. He's just an atheist and might crack a joke or two about religion. For example, as his hit character Stewie in , MacFarlane says:",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Seth MacFarlane",
      "sourceUrl": "http://www.celebatheists.com/wiki/Seth_MacFarlane",
      "text": "Stay away from the church. In the battle over science vs. religion, science offers credible evidence for all the serious claims it makes. The church says, 'Oh, it's right here in this book, see? The one written by people who thought the sun was magic?' I for one would like to see some proof that there is a God. And if you say 'a baby's smile' I'm going to kick you right in the stomach.",
      "type": "quote",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "Certainly not the \\"Family Values\\" of the Republican party",
      "type": "heading",
    },
    Object {
      "sourceTitle": "Seth MacFarlane’s Federal Campaign Contribution Report",
      "sourceUrl": "http://www.newsmeat.com/celebrity_political_donations/Seth_MacFarlane.php",
      "text": "MacFarlane fits easily into the liberal designation. First of all, and like most American atheists, forced to choose between two and only two parties, MacFarlane is a Democrat. He's given over $92,000 to the Democratic Party and Democrat-associated special interest groups.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Seth MacFarlane Speaks for Obama, Columbus, OH, Oct 2008",
      "sourceUrl": "http://www.youtube.com/watch?v=ZqXneQkWXQ8",
      "text": "He was an Obama supporter during the 2008 presidential campaign",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Seth MacFarlane Says Romney Can Beat Obama",
      "sourceUrl": "http://www.youtube.com/watch?v=ryPhPQ2wrKQ",
      "text": "and expressed concern over the possibility of Mitt Romney beating Obama in the 2012 elections.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "To add to his list of liberal credentials, MacFarlane has been a staunch supporter of gay rights and came out against Arizona's draconian immigration law. MacFarlane is in the camp that progressive society will march forward despite conservative opposition, saying:",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Seth MacFarlane: Arizona Immigration, Gay Marriage",
      "sourceUrl": "http://www.mediaite.com/online/seth-macfarlane-on-arizonas-immigration-law-gay-marriage-and-todays-culture/",
      "text": "[The] civil-rights battle will be won eventually. They'll just have to sulk in a corner and accept the fact that gays will be able to marry each other. Every civil-rights conflict—black people, women—comes to an end in the right way.",
      "type": "quote",
    },
    Object {
      "sourceTitle": "Seth MacFarlane Trivia & Quotes",
      "sourceUrl": "http://www.tv.com/people/seth-macfarlane/trivia/",
      "text": "Oh, and he thinks pot should be legal.",
      "type": "sentence",
    },
  ],
  "notablePerson": Object {
    "author": "Tom Kershaw",
    "lastUpdatedOn": 2012-05-07T21:00:00.000Z,
    "name": "Seth MacFarlane",
    "relatedPeople": Array [
      Object {
        "name": "Bill Maher",
        "slug": "bill-maher",
      },
      Object {
        "name": "George Clooney",
        "slug": "george-clooney",
      },
      Object {
        "name": "Ben Stiller",
        "slug": "ben-stiller",
      },
      Object {
        "name": "Hugh Laurie",
        "slug": "hugh-laurie",
      },
    ],
    "summary": Object {
      "politicalViews": "MacFarlane is a big supporter of the Democratic Party and Obama. And in this case, big means big money. He's a liberal who supports gay marriage, liberal immigration policy, and the legalization of marijuana.",
      "religion": "MacFarlane was raised a Catholic, but turned that into a joke by the time he was 11 years-old. Now, he proudly claims atheism.",
    },
    "tags": Array [
      "actor",
      "atheist",
      "catholic",
      "comedian",
      "democrat",
      "director",
      "liberal",
      "producer",
      "screenwriter",
      "singer",
      "voice-actor",
      "white",
    ],
  },
}
`;

exports[`tom-hanks 1`] = `
Object {
  "content": Array [
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "Tom Hanks was born in Concord, California.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "His religious beliefs are about as diverse as the characters he has played. Hanks' childhood was a hodgepodge of divorce and step-parents, each with their own views. Hanks has identified with everything from Mormonism to his most recent and enduring faith–Greek Orthodox. In his own words:",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Mystagogy: The Orthodoxy of Tom Hanks",
      "sourceUrl": "http://www.johnsanidopoulos.com/2009/05/orthodoxy-of-tom-hanks.html",
      "text": "The major religion I was exposed to in the first 10 years of my life was Catholicism. My stepmother became a Mormon. My aunt, whom I lived with for a long time, was a Nazarene, which is kind of ultra-super Methodist, and in high school, all my friends were Jews. For years I went to Wednesday-night Bible studies with my church group. So I had this peripatetic overview of various faiths, and the one thing I got from that was the intellectual pursuit involved. There was a lot of great stuff to think about.",
      "type": "quote",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "Hanks' second and current wife, Rita Wilson, was raised a member of the Greek Orthodox Church and Hanks has taken on her religion.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Tom Hanks",
      "sourceUrl": "http://www.beliefnet.com/Celebrity-Faith-Database/H/Tom-Hanks.aspx",
      "text": "Hanks is said to be a deeply religious man and goes to church regularly.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Mystagogy: The Orthodoxy of Tom Hanks",
      "sourceUrl": "http://www.johnsanidopoulos.com/2009/05/orthodoxy-of-tom-hanks.html",
      "text": "Hanks has said that his greatest motivation for being a church-goer is to come in contact with the great unanswered questions mankind has always asked, questions that, for the most part, only religion attempts to answer.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "Politics is just like a box of chocolates…",
      "type": "heading",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "Politically, Hanks falls squarely in the Democrat/liberal camp, like the vast majority of his Hollywood counterparts. He's stuck up for quite a few liberal causes like gay marriage and the environment, and has a large collection of electric and hybrid cars.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Does Tom Hanks Hate Mormons?",
      "sourceUrl": "http://mormonmatters.org/2009/03/11/does-tom-hanks-hate-mormons/",
      "text": "He even slandered his old church–The Church of Jesus Christ of Latter Day Saints (Mormons) when they donated money to advance proposition 8 in California, effectively banning gay marriage.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Proposition 8: Tracking the Money",
      "sourceUrl": "http://www.latimes.com/news/local/la-moneymap,0,2198220.htmlstory",
      "text": "However, he quickly rescinded his statement but fought fire with fire, contributing his star power to raising over $44 million to defeat proposition 8–which passed anyway.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Beware: Celebrity Endorsement Video by Tom Hanks",
      "sourceUrl": "http://www.myspace.com/video/vid/33546997",
      "text": "Hanks backed Obama in 2008, posting a video on his Myspace page, explaining his position.",
      "type": "sentence",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "In 2012, Hanks forcefully said he would back Obama once again and, despite widespread opinion to the opposite effect, said Obama had exceeded his expectations, saying:",
      "type": "sentence",
    },
    Object {
      "sourceTitle": "Tom Hanks: I’m Voting for Obama Again",
      "sourceUrl": "http://www.politico.com/blogs/click/0611/Tom_Hanks_Im_voting_for_Obama_again.html",
      "text": "If you would have told me a few years ago that 'don't ask, don't tell' would be repealed and about a billion jobs at General Motors and Chrysler would have been saved because the president was smart enough and strong enough and bold enough to do so, I would have said, 'Wow. That's a good president, I think I'll vote for him again.'",
      "type": "quote",
    },
    Object {
      "sourceTitle": undefined,
      "sourceUrl": undefined,
      "text": "He doesn't seem to acknowledge the billions upon billions of taxpayer dollars that were funneled to some of the country's richest men during that bailout, but hey, that's what the PR machine is all about.",
      "type": "sentence",
    },
  ],
  "notablePerson": Object {
    "author": "Tom Kershaw",
    "lastUpdatedOn": 2012-04-04T21:00:00.000Z,
    "name": "Tom Hanks",
    "relatedPeople": Array [
      Object {
        "name": "George Clooney",
        "slug": "george-clooney",
      },
      Object {
        "name": "Seth MacFarlane",
        "slug": "seth-macfarlane",
      },
      Object {
        "name": "Al Pacino",
        "slug": "al-pacino",
      },
      Object {
        "name": "Kevin Smith",
        "slug": "kevin-smith",
      },
    ],
    "summary": Object {
      "politicalViews": "Hanks is a hardcore Democrat and liberal, throwing his weight behind gay marriage, alternative energy, environmentalism, and Obama.",
      "religion": "Hanks has run the gamut of Christian denominations from Mormonism to Catholicism but seems to have settled on Greek Orthodox.",
    },
    "tags": Array [
      "actor",
      "catholic",
      "democrat",
      "director",
      "greek-orthodox",
      "liberal",
      "mormon",
      "producer",
      "religious",
      "white",
    ],
  },
}
`;
